drivers/adc/adm1177/adm1177.c

```diff
@@ -0,0 +1,315 @@
+ * @brief Free the resources allocated by adm1177_init().
+ *
+ * @param device - The device structure.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_remove(struct adm1177_dev *device)
+{
+	int ret;
+
+	ret = no_os_i2c_remove(device->i2c_desc);
+
+	no_os_free(device);
+
+	return ret;
+}
+
+/**************************************************************************//**
+ * @brief - In the write command byte operation, the master device sends a
+ *          command byte to the slave device.
+ *
+ * @param device - The device structure.
+ *
+ * @param cmd - The command byte indicates which operation to be performed.
+ *
+ * @param reg_address - In case of a write extended command byte operation
+ * 			one of the ADM1177_REG_ALERT_EN, ADM1177_REG_ALERT_TH
+ * 			or ADM1177_REG_CONTROL will be used.
+ * 			In case of a write command byte operation reg_address
+ * 			will be set to 0.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_write(struct adm1177_dev *device, uint8_t cmd, uint8_t reg_address)
+{
+	if(reg_address & ADM1177_EXT_CMD_MASK) {
+		uint8_t data[2] = {reg_address, cmd};
+
+		return no_os_i2c_write(device->i2c_desc, data, 2, 1);
+	} else {
+		if(cmd == ADM1177_CMD_I_CONT || cmd == ADM1177_CMD_I_ONCE) {
+			device->last_command = ADM1177_CURRENT_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd == ADM1177_CMD_V_CONT || cmd == ADM1177_CMD_V_ONCE) {
+			device->last_command = ADM1177_VOLTAGE_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if((cmd == (ADM1177_CMD_I_CONT | ADM1177_CMD_V_CONT)) ||
+		    (cmd == (ADM1177_CMD_I_ONCE | ADM1177_CMD_V_ONCE))) {
+			device->last_command = ADM1177_VOLTAGE_AND_CURRENT_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd == ADM1177_CMD_STATUS_RD) {
+			device->last_command = ADM1177_STATUS_EN;
+			return no_os_i2c_write(device->i2c_desc, &cmd, 1, 1);
+		}
+		if(cmd & ADM1177_CMD_VRANGE) {
+			device->vrange = ADM1177_VRANGE_HIGH;
+		} else {
+			device->vrange = ADM1177_VRANGE_LOW;
+		}
+		return -EINVAL;
+	}
+}
+
+/**************************************************************************//**
+ * @brief - Reads the status byte from the ADC if requested and transmits it
+ * 	    through the status_byte pointer.
+ *
+ * @param device - The device structure.
+ *
+ * @param status_byte - The pointer through which the status byte is
+ * 			transmitted.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+******************************************************************************/
+int adm1177_read_status(struct adm1177_dev *device, uint8_t *status_byte)
+{
+	if(device->last_command == ADM1177_STATUS_EN) {
+		if(status_byte == NULL)
+			return -EINVAL;
+		int ret;
```

Comment:
move variable declarations at the beginning of function, I'm pretty sure this is in the style guide

Reply:
ok.